src/main/java/io/cdap/plugin/snowflake/sink/batch/SnowflakeOutputFormat.java

```diff
@@ -78,8 +79,19 @@ public void commitJob(JobContext jobContext) throws IOException {
         String destinationStagePath = conf.get(DESTINATION_STAGE_PATH_PROPERTY);
 
         SnowflakeSinkAccessor snowflakeAccessor = new SnowflakeSinkAccessor(config);
-        snowflakeAccessor.populateTable(destinationStagePath);
-        snowflakeAccessor.removeDirectory(destinationStagePath);
+        try {
+          snowflakeAccessor.populateTable(destinationStagePath);
+          snowflakeAccessor.removeDirectory(destinationStagePath);
+        } catch (IOException e) {
+          String errorMessage = String.format(
+            "Error committing job: Failed to populate table and remove directory at path '%s' with message: %s .",
+            destinationStagePath, e.getMessage()
+          );
+          String errorReason = String.format("Error committing job: Failed to populate table and " +
+            "remove directory at path '%s'.", destinationStagePath);
+          throw ErrorUtils.getProgramFailureException(new ErrorCategory(ErrorCategory.ErrorCategoryEnum.PLUGIN),
+            errorReason, errorMessage, ErrorType.SYSTEM, true, e);
+        }
```

Comment:
Instead of catching `IOException` thrown from `SnowflakeAccessor#runSQL`, wrap `SQLException` from here to `ProgramFailureException`, so you would not need to make any change here: https://github.com/data-integrations/snowflake-plugins/blob/1d9cacd36d507454c9d25efbafd88e6803852009/src/main/java/io/cdap/plugin/snowflake/common/client/SnowflakeAccessor.java#L63-L65